lib/internal/warn.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+// A process Warning is similar in structure to an Error except that they
+// are emitted by the process.on('warn') event rather than being thrown.
+
+const _message = Symbol('message');
+const _name = Symbol('name');
+const prefix = `(${process.release.name}:${process.pid}) `;
+
+class Warning {
+  constructor(name, message, constructor) {
+    this[_name] = name;
+    this[_message] = `${prefix}${message}`;
+    Error.captureStackTrace(this, constructor || Warning);
+  }
+
+  get message() {
+    return this[_message];
+  }
+
+  get name() {
+    return this[_name];
```

Comment:
Why not use `_name` and `_message` properties? This seems confusing and unnecessary?

(It also makes inspecting the warnings difficult?)